labs/lab-4-jquery/tables.js

```diff
@@ -0,0 +1,34 @@
+class Table {
+  constructor(number) {
+    this.number = number;
+  }
+
+  reserveTable(name, partySize) {
+    this.reservationName = name;
+    this.partySize = partySize;
+  }
+
+  render() {
+    var button = $(`<button class="table" id=table-${this.number}>${this.number + 1}</button>`)
+    $(".floor").append(button);
+  }
+}
+
+const numberOfTables = 9;
+var tables = [];
+var currentTable;
+
+$(document).ready(function() {
+  for (i = 0; i < numberOfTables; i++) {
+    tables.push(new Table(i));
+    tables[i].render();
+  }
+
+  // Clicking on the table
+  $(".table").click(function() {
```

Comment:
This is cute and clever and you can totally do it for this lab. I wouldn't have thought of doing it this way. Nice work!

I'd encourage you to take a look at the example code for this lab I pushed to master; in particular, see how each Table takes care of itself; aside from triggering rerender, Table is self-contained.

Breaking things into self-sufficient components like that is both a good practice generally, makes it easier to work on a team, AND it will mean you're well-prepared conceptually for Angular.